Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonAnimationManager.cs

```diff
@@ -0,0 +1,149 @@
+        PlayerManager.Instance.GetOnHarpoonStartReloadEvent().AddListener(ReloadHarpoonAnimation);
+    }
+
+    /// <summary>
+    /// Unsubscribe to the needed events
+    /// </summary>
+    private void UnsubscribeToEvents()
+    {
+        PlayerManager.Instance.GetOnHarpoonFocusStartEvent().RemoveListener(StartFocusingAnimation);
+        PlayerManager.Instance.GetOnHarpoonFocusEndEvent().RemoveListener(StopFocusingAnimation);
+        PlayerManager.Instance.GetOnHarpoonFiredEvent().RemoveListener(StartFiringAnimation);
+        PlayerManager.Instance.GetOnHarpoonRestockCompleteEvent().RemoveListener(ReloadFromEmptyAnimation);
+        PlayerManager.Instance.GetOnHarpoonStartReloadEvent().RemoveListener(ReloadHarpoonAnimation);
+    }
+
+    /// <summary>
+    /// Changes the animation to the focusing animation
+    /// </summary>
+    private void StartFocusingAnimation()
+    {
+        _animator.SetTrigger(_FOCUS_START_ANIM);
+        _animator.ResetTrigger(_FOCUS_END_ANIM);
+    }
+
+    /// <summary>
+    /// Changes the animation from the focusing animation back to the idle animation
+    /// </summary>
+    private void StopFocusingAnimation()
+    {
+        _animator.SetTrigger(_FOCUS_END_ANIM);
+    }
+
+    /// <summary>
+    /// Changes the animation from the focusing animation to the firing animation
+    /// </summary>
+    private void StartFiringAnimation()
+    {
+        _animator.SetTrigger(_FIRE_ANIM);
+
+        //Check to see if we are out of ammo
+        //If we are, start the empty ammo animation
+        if (HarpoonGun.Instance.GetHarpoonFiringState() != HarpoonGun.EHarpoonFiringState.Ready)
+        {
+            if (HarpoonGun.Instance.GetReserveAmmo() == 0)
+            {
+                _animator.SetTrigger(_AMMO_EMPTY_ANIM);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Changes the animation to the reloading animation
+    /// </summary>
+    private void ReloadHarpoonAnimation()
+    {
+        _animator.SetTrigger(_RELOAD_READY_ANIM);
+    }
+    
+    /// <summary>
+    /// Separate animation change after restocking on ammo
+    /// If we went from completely empty to full, play the reloading animation
+    /// </summary>
+    /// <param name="ammoRestocked"> The amount of ammo that was restocked </param>
+    private void ReloadFromEmptyAnimation(int ammoRestocked)
+    {
+        if (ammoRestocked == HarpoonGun.Instance.GetMaxAmmo())
+        {
+            _animator.SetTrigger(_RELOAD_READY_ANIM);
+        }
+    }
+
+    /// <summary>
+    /// Coroutine that checks to see if the player is near a wall
+    /// Used to start and stop the holstering animation
+    /// </summary>
+    private IEnumerator CheckForWall()
+    {
+        while(true)
+        {
+            if (Physics.Raycast(transform.position, transform.parent.forward, _WALL_CHECK_DIST))
+            {
+                _animator.SetTrigger(_AT_WALL_ANIM);
+                _animator.ResetTrigger(_NOT_WALL_ANIM);
```

Comment:
I feel these could also be a bool rather than 2 triggers, but this is a recommendation not a requirement